test/membrane_ice_plugin/turn_cleaner_test.exs

```diff
@@ -36,4 +37,33 @@ defmodule Membrane.ICE.TURNCleanerTest do
     assert %{specs: 0, active: 0, supervisors: 0, workers: 0} ==
              DynamicSupervisor.count_children(turn_cleaner_sup)
   end
+
+  defp count_children_sync(sup) do
+    my_pid = self()
+
+    pid =
+      spawn(fn ->
+        send(my_pid, count_children_rec(sup))
+      end)
+
+    receive do
+      {:ok, value} ->
+        value
+    after
+      1000 ->
+        Process.exit(pid, :kill)
+        DynamicSupervisor.count_children(sup)
+    end
+  end
+
+  defp count_children_rec(sup) do
+    case DynamicSupervisor.count_children(sup) do
+      %{specs: 0, active: 0, supervisors: 0, workers: 0} ->
+        Process.sleep(10)
+        count_children_rec(sup)
+
+      non_zero_res ->
+        {:ok, non_zero_res}
+    end
+  end
```

Comment:
I think, that since we have `Membrane.ResourceGuard`, we can delete `Membrane.ICE.TURNCleaner` and it's tests and use `ResoruceGuard` in `ICE.Endpoint.handle_setup/2`

Reply:
In another PR.

Reply:
Do you mean you will make this PR?

Reply:
All what is needed to be done, is to 
1) put `Membrane.ResourceGuard.register(ctx.resource_guard, fn -> Membrane.ICE.Utils.stop_integrated_turn(turn) end)` in places of calls to `Membrane.ICE.TURNCleaner.start_under/3`
2) delete `TURNCleaner` module and its test
3) delete starting `Membrane.ICE.TURNCleaner.Sup` in application module and `turn_cleaner_sup` from `ICE.Endpoint` `def_options`